index.js

```diff
@@ -1,58 +1,158 @@
+const cwd = __dirname;
 
-const process = (options = {}) => (source, filename) => {
-	const { preprocess, debug, compilerOptions, noStyles } = options;
-
-  // strip out <style> tags to prevent errors with node-sass.
-  const normalized = noStyles !== false ? source.replace(styleRegex, '') : source;
-  
-	let preprocessed;
-
-	if (preprocess) {
-		svelte
-			.preprocess(normalized, preprocess || {}, {
-				filename
-			})
-			.then((result) => (preprocessed = result.code));
-
-		deasync.loopWhile(() => !preprocessed);
-	} else {
-		preprocessed = normalized;
-	}
-
-	const compiled = svelte.compile(preprocessed, {
-		filename,
-		css: false,
-		// Allow tests to set component props.
-		accessors: true,
-		// Debugging and runtime checks
-		dev: true,
-		// Emit CommonJS that Jest can understand.
-		format: 'cjs',
-		...compilerOptions
-	});
-
-	// Fixes the '_Sample.default is not a constructor' error when importing in Jest.
-	const esInterop =
-		'Object.defineProperty(exports, "__esModule", { value: true });';
-
-	const code = compiled.js.code + esInterop;
-
-	if (debug) {
-		console.log(code);
-	}
-
-	return {
-		code,
-		map: compiled.js.map
-	};
-};
+// Fixes the '_Sample.default is not a constructor' error when importing in Jest.
+const esInterop = 'Object.defineProperty(exports, "__esModule", { value: true });';
+
+module.exports = {
+
+    /**
+     * Create the transformer
+     * @param {Object} transformOptions
+     * @param {?Object} transformOptions.preprocess - the svelte preprocess config, if not, no preprocessing
+     * @param {Object} transformOptions.compilerOptions - options for the Svelte compiler
+     * @param {boolean} transformOptions.noStyles - to just remove styles from svelte sources
+     * @returns {Object} the transformer
+     */
+    createTransformer(transformOptions) {
+        const { preprocess: preprocessConfig, compilerOptions, noStyles } = transformOptions;
+
+        /**
+         * Normalize svelte sources (strip out styles if configured)
+         * @param {string} source
+         * @returns {string} the normalized source
+         */
+        function getNormalizedSource(source) {
+            return noStyles !== false ? source.replace(styleRegex, '') : source;
+        }
+
+        /**
+         * Compile the svelte source file
+         * @param {string} source
+         * @param {string} filename
+         * @param {boolean} supportsStaticESM - if true we compiled to ESM
+         * @returns {Object} the compilation object
+         */
+        function svelteCompile(source, filename, supportsStaticESM = false) {
+            return svelte.compile(
+                source,
+                Object.assign(
+                    {
+                        filename,
+                        css: false,
+                        // Allow tests to set component props.
+                        accessors: true,
+                        // Debugging and runtime checks
+                        dev: false,
```

Comment:
Did you change `true` -> `false` for a reason?

Reply:
We set `dev: false` in all our projects using the compiler options. It seems to be a good default.